internal/pkg/git/git.go

```diff
@@ -23,6 +23,45 @@ func Available() bool {
 	return err == nil
 }
 
+func CheckoutWholeRepository(repo model.TemplateRepository, logger log.Logger) (filesystem.Fs, string, error) {
```

Comment:
Mozno nejak vylepsit naming?, mame tu:
`CheckoutWholeRepository` a `CheckoutTemplateRepository`, ciastocne sa prekryvaju.

Mozno: `CheckoutTemplateRepositoryPartal`, `CheckoutTemplateRepositoryFull`?

Najlepsie by mozno bolo, ak by to bola jedna funkcia, `CheckoutTemplateRepository` a mala by `CheckoutOptions`:
```
type CheckoutOptions struct {
  Partial bool                 // ...
  CopyToMemory bool  // ...
}
```

Pribudli by tam nejake podmieky, ... no nebol by duplicitny kod, a ak tam budu nejake komentare, ... tak by sa to mohlo dobre citat, ze v com je rozdiel a ako to funguje.

Reply:
Jo, naming upravím. 👍

Nad tím jsem dost dumal jestli by to nešlo dát do jedné funkce ale moc nevím, těch rozdílů je tam dost: - vytvořený temp dir se deferem maže jen u parciálního klonu (protože se převede do paměti ale u full klonu zůstane na disku)
- jsou různý parametry u `git clone`
- u parciálního se provede `git sparse-checkout`
  - načte se manifest a hledá se v něm to konkrétní repo
  - checkoutne se repo
  - vytvoří se memory fs a nakopíruje do něj ten temp
🤔 ale jak jsem to teď rozepdsal, tak už to asi nevypadá tak hrozně. 😃 Zkusím to.